src/Symfony/Component/Notifier/Bridge/Slack/Tests/SlackOptionsTest.php

```diff
@@ -0,0 +1,185 @@
+    {
+        $this->assertSame($expected ?? $options, (new SlackOptions($options))->toArray());
+    }
+
+    public function toArrayProvider(): iterable
+    {
+        yield 'empty is allowed' => [
+            [],
+            [],
+        ];
+
+        yield 'always unset recipient_id' => [
+            ['recipient_id' => '42'],
+            [],
+        ];
+
+        yield 'blocks containing 1 divider block' => [
+            [
+                'blocks' => [
+                    $block = new SlackDividerBlock(),
+                ],
+            ],
+            [
+                'blocks' => [
+                    $block,
+                ],
+            ],
+        ];
+    }
+
+    public function toArraySimpleOptionsProvider(): iterable
+    {
+        yield [['as_user' => true]];
+        yield [['icon_emoji' => 'foo']];
+        yield [['icon_url' => 'https://symfony.com']];
+        yield [['link_names' => true]];
+        yield [['mrkdwn' => true]];
+        yield [['parse' => 'bar']];
+        yield [['unfurl_links' => true]];
+        yield [['unfurl_media' => true]];
+        yield [['username' => 'baz']];
+    }
+
+    /**
+     * @dataProvider getRecipientIdProvider
+     */
+    public function testGetRecipientId(?string $expected, SlackOptions $options)
+    {
+        $this->assertSame($expected, $options->getRecipientId());
+    }
+
+    public function getRecipientIdProvider(): iterable
+    {
+        yield [null, new SlackOptions()];
+        yield [null, (new SlackOptions(['recipient_id' => null]))];
+        yield ['foo', (new SlackOptions())->recipient('foo')];
+        yield ['foo', (new SlackOptions(['recipient_id' => 'foo']))];
+    }
+
+    /**
+     * @dataProvider setProvider
+     *
+     * @param mixed $value
+     */
+    public function testSet(string $method, string $optionsKey, $value)
+    {
+        $options = (new SlackOptions())->$method($value);
+
+        $this->assertSame($value, $options->toArray()[$optionsKey]);
+    }
+
+    public function setProvider(): iterable
+    {
+        yield ['asUser', 'as_user', true];
+        yield ['iconEmoji', 'icon_emoji', 'foo'];
+        yield ['iconUrl', 'icon_url', 'https://symfony.com'];
+        yield ['linkNames', 'link_names', true];
+        yield ['mrkdwn', 'mrkdwn', true];
+        yield ['parse', 'parse', 'bar'];
+        yield ['unfurlLinks', 'unfurl_links', true];
+        yield ['unfurlMedia', 'unfurl_media', true];
+        yield ['username', 'username', 'baz'];
+    }
+
+    public function testSetBlock()
+    {
+        $options = (new SlackOptions())->block(new SlackDividerBlock());
+
+        $this->assertSame([['type' => 'divider']], $options->toArray()['blocks']);
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testChannelMethodRaisesDeprecation()
+    {
+        $this->expectDeprecationMessage('Since symfony/slack-notifier 5.1: The "channel()" method is deprecated, use "recipient()" instead.');
```

Comment:
```suggestion
        $this->expectDeprecation('Since symfony/slack-notifier 5.1: The "channel()" method is deprecated, use "recipient()" instead.');
```

Also, you need to use `ExpectDeprecationTrait`.

Reply:
Ah didn't know that, thanks